api/src/client/dynamics/searchBusinessAddressesForMatch.ts

```diff
@@ -0,0 +1,31 @@
+import { LicenseSearchAddress } from "@shared/license";
+import { BusinessIdAndLicenseSearchAddresses } from "./types";
+
+export const searchBusinessAddressesForMatch = (
+  businessIdsAndLicenseAddresses: BusinessIdAndLicenseSearchAddresses[],
+  nameAndAddress: LicenseSearchAddress
+): string => {
+  const result: string[] = [];
+
+  const isAddressMatch = (
+    address: LicenseSearchAddress,
+    userInputtedAddress: LicenseSearchAddress
+  ): boolean => {
+    return (
+      userInputtedAddress.addressLine1 === address.addressLine1 &&
+      address.zipCode === userInputtedAddress.zipCode
+    );
+  };
+
+  for (const { businessId, addresses } of businessIdsAndLicenseAddresses) {
+    for (const address of addresses) {
+      if (isAddressMatch(address, nameAndAddress)) {
+        result.push(businessId);
+        break;
```

Comment:
Should this break be here? If we want to make sure that we're only getting one match, wouldn't we want to iterate over the whole list? This would cause us to exit the for loop on the first match, right?